### ISSUE TITLE
Memory leak on binary view after using the debugger if the input view was rebased during launch

### ISSUE THREAD

**Issue by xusheng6** [Vector35/debugger#546]
This happens in 4.0.4898-dev (2e55170c). I noticed this while troubleshooting https://github.com/Vector35/debugger/issues/545

This does not happen on macOS

**Comment by xusheng6** [comment]
This even exists before stable 3.5, probably introduced earlier

**Comment by xusheng6** [comment]
This is pretty puzzling -- the two leaked views are the debugger null view and the raw view. The PE view and the debugger view are properly freed. Among them, the debugger null view gets referenced twice, and it never gets freed. The ref for the raw view goes up and down, and eventually there are 3 refs to it

**Comment by xusheng6** [comment]
OK so this has nothing to do with Windows -- as long as the file was rebased during the creation of the debugger view, the leak is observed. The windows test binaries happen to all requrie a rebase so we observe the leak. The macOS test binaries I used (helloworld), is not PIE so there was no such rebase.

**Comment by xusheng6** [comment]
My bad, we are actually leaking the original raw and PE view, which is probably caused by the rebasing. Note, right now, rebasing itself no longer causes a leak. So it is probably caused by a combination of things

```
[Default] making view: Raw, this: 0x23f24421800
[Default] making view: PE, this: 0x23f24422400
[Default] making view: Debugger Null, this: 0x23fd7ac0000
[Default] making view: Debugger, this: 0x23fd7ac0c00
[Default] making view: PE, this: 0x23fcd851e00
[Default] freeing view: PE, this: 0x23fcd851e00
[Default] making view: PE, this: 0x23fffe10000
[Default] freeing view: Debugger, this: 0x23fd7ac0c00
[Default] freeing view: Debugger Null, this: 0x23fd7ac0000
[Default] freeing view: PE, this: 0x23fffe10000
```

**Comment by plafosse** [comment]
Fixed in 4.0.4902
